src/main.rs

```diff
@@ -186,7 +188,7 @@ fn main() {
         }
         let updated = march(&camera, &objects, &lights);
         for i in 0..updated.len() {
-            pixels[i] = (updated[i] * 255.) as u8;
+            pixels[i] = updated[i];
```

Comment:
you can replace manual copying of slices with `pixels[..updated.len()].clone_from_slice(&updated[..])`

Reply:
Thanks, I'll try that.